pygmt/tests/test_rose.py

```diff
@@ -0,0 +1,298 @@
+"""
+Tests for rose.
+"""
+import numpy as np
+from pygmt import Figure
+from pygmt.helpers.testing import check_figures_equal
+
+
+@check_figures_equal()
+def test_rose_data_file():
+    """
+    Test supplying a data file containing a list of fracture lengths and
+    directions as digitized from geological maps to the data argument (lengths
+    are stored in the second column, directions in the first, specify via
+    columns).
+
+    Passing in R as list and string.
+    """
+
+    fig_ref, fig_test = Figure(), Figure()
+
+    fig_ref.rose(
+        data="@fractures_06.txt",
+        region="0/1/0/360",
+        sector=15,
+        diameter="5.5c",
+        color="blue",
+        frame=["x0.2g0.2", "y30g30", "+glightgray"],
+        columns=[1, 0],
+        pen="1p",
+        norm="",
+        scale=0.4,
+    )
+
+    fig_test.rose(
+        data="@fractures_06.txt",
+        R=[0, 1, 0, 360],
+        A=15,
+        JX="5.5c",
+        G="blue",
+        B=["x0.2g0.2", "y30g30", "+glightgray"],
+        i=[1, 0],
+        W="1p",
+        S=True,
+        Z=0.4,
+    )
+
+    return fig_ref, fig_test
```

Comment:
Hi @michaelgrund, really sorry for the long wait and back and forth trouble, but would you be able to refactor this to use the simpler `@pytest.mark.mpl_image_compare`? We're deciding to store the PNG base images via `dvc` on [DAGsHub](https://dagshub.com/GenericMappingTools/pygmt/branches) now, see specific details at https://github.com/GenericMappingTools/pygmt/issues/963#issuecomment-801606095.

Reply:
No problem @weiji14 😉. Hopefully, I have time this weekend to make the changes regarding  ``@pytest.mark.mpl_image_compare``. Also need to do some other adjustments.

Reply:
Cool, tell me when you've got a DAGsHub account set up (can be linked to your GitHub user) and I'll add you as a collaborator so that you can do `dvc push` to the remote repository at https://dagshub.com/GenericMappingTools/pygmt.

Reply:
ok, added you in :smile:

Reply:
Will work on the tests as soon as possible 😉 @weiji14.